db/downloader/downloader.go

```diff
@@ -1330,19 +1329,23 @@ func (d *Downloader) state() DownloaderState {
 
 // Currently only called if not all torrents are complete.
 func (d *Downloader) logStats() {
-	bytesDone := d.stats.BytesCompleted
-	percentDone := float32(100) * (float32(bytesDone) / float32(d.stats.BytesTotal))
-	remainingBytes := d.stats.BytesTotal - bytesDone
+	d.lock.RLock()
+	// This is set externally. Everything else here is only modified by the caller.
```

Comment:
```

	d.lock.Lock()
	d.stats = stats
	d.lock.Unlock()
```
write happening by mutex. if read happening without mutex - then it's very weird.